docs/ops/generation/Range_4.md

```diff
@@ -0,0 +1,124 @@
+## Range<a name="Range"></a>
+
+**Versioned name**: *Range-4*
+
+**Category**: Generation
+
+**Short description**: *Range* operation generates a sequence of numbers according input values [start, stop) with a step.
+
+**Attributes**:
+
+* *output_type*
+
+  * **Description**: the output tensor type
+  * **Range of values**: any numeric type
+  * **Type**: string
+  * **Default value**: None
+  * **Required**: *Yes*
+
+**Inputs**:
+
+* **1**: "start" - A scalar of type T1. **Required.**
+* **2**: "stop" - A scalar of type T2. **Required.**
+* **3**: "step" - A scalar of type T3. If `step` is equal to zero after casing to `output_type`, behavior is undefined. **Required.**
+
+**Outputs**:
+
+* **1**: A tensor with type specified by attribute *output_type*.
+
+**Types**
+
+* *T1*, *T2*, *T3*: any numeric type.
+
+**Detailed description**:
+
+*Range* operation generates a sequence of numbers starting from the value in the first input (`start`) up to but not including the value in the second input (`stop`) with a `step` equal to the value in the third input, according to the following formula:
+
+For a positive `step`:
+
+\f[
+start<=val[i]<stop,
+\f]
+
+for a negative `step`:
+
+\f[
+start>=val[i]>stop,
+\f]
+
+to lower the impact of floating point rounding errors the i-th element is calculated by the following formula:
+
+\f[
+val[i]=start+i*step.
+\f]
+
+The calculations are done after casting all values to `output_type`, while the number of elements is calculated in the floating point type according to the following formula:
```

Comment:
I am confused with "The calculations are done after casting ...". It seems about TF.

Reply:
The difference between tf and pytorch is in length calculation, the individual values are calculated similarly.

Reply:
Are they always calculate the number of elements in floating point type?

Reply:
Yes, as far as I understood.